stereotype/fields/base.py

```diff
@@ -124,8 +125,8 @@ def make_validated_config(self) -> _ValidatedFieldConfig:
                 self.native_validate, self.validator_method, self.validators)
 
     def make_output_config(self) -> _OutputFieldConfig:
-        return (self.name, self.serializable, self.to_primitive if not self.atomic else None,
-                self.to_primitive_name, self.hide_none, self.hide_empty, self.empty_value)
+        return (self.name, self.serializable, self.to_primitive, self.to_primitive_name,
```

Comment:
This removes an intentional optimization, why?

The performance degradation of conversion + serialization after removing this optimization in my somewhat real-world benchmark is around 8%, so it would need to have a good reason...

Reply:
Ah, I see, it's because of `PrivateStrField`. Hmm, maybe instead of using the `atomic` flag for determining whether `to_primitive` has to be called for a field, stereotype could try to determine whether the method was overridden?

Reply:
At first I was going to say just add `atomic = False` to `PrivateStrField`, but it makes sense users of the library might want to inherit from the builtin atomic types and customize the serialization logic...

Reply:
BTW, if you don't feel like doing that change, I can totally do it myself :)

Reply:
I also added it to `DictField` and `ListField` where I missed this check before and added tests for both of those cases.